libs/perun/pipes/src/lib/manageable-entities.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { GuiAuthResolver } from '@perun-web-apps/perun/services';
+import { PerunBean } from '@perun-web-apps/perun/openapi';
+import { AuthPrivilege } from '@perun-web-apps/perun/models';
+
+@Pipe({
+  name: 'manageableEntities',
+})
+export class ManageableEntitiesPipe implements PipeTransform {
+  constructor(private authResolver: GuiAuthResolver) {}
+
+  transform(entities: PerunBean[], role: string): PerunBean[] {
+    return entities.filter((e) => this.canSetRoleForEntity(e, role));
+  }
+
+  private canSetRoleForEntity(entity: PerunBean, role: string): boolean {
+    const manageRights = new Map<string, AuthPrivilege>();
+    this.authResolver.setRolesAuthorization([role], entity, manageRights);
```

Comment:
There are several problems that relate to the nested `function fetchAllRelatedObjects(...)` and the "Facility" object. Firstly at a `roles-page.component.ts` you put as an argument Rich variants of objects, but the Rich variant for the `Facility` is `EnrichedFacility` which doesn't extend PerunBean and has a facility as a nested property. Secondly, if you correctly parse the Facility object, the parameter of the `fetchAllRelatedObjects(...)` method is `Entity[]` type (Vo & Group & Resource & Member & User), so we probably want to add also Facility to this `Entity` type.

You can test this problem when you want to check some facility checkbox under the facility admin expansion panel.

Reply:
Enriched facility should now be parsed at all places.